Display.lua

```diff
@@ -535,6 +535,12 @@ function WarpDeplete:SetObjectives(objectives)
 end
 
 function WarpDeplete:UpdateObjectivesDisplay()
+  if not self.db.profile.showObjectives then 
+    for i, boss in ipairs(self.objectivesState) do
```

Comment:
You can iterate directly through the objective texts here, instead of the objectives state:

```lua
for _, obj in pairs(self.frames.root.objectiveTexts) do obj:SetText("") end
```